### ISSUE TITLE
Performance comparison between v5 and v6

### ISSUE THREAD

**Issue by aight8** [erikras/redux-form#851]
The main goal of redux form v6 was mainly the performance issues because of redux subscription point. I'm wondering if this problem is really solved in v6, i dont have a big form setup so I can't admit that.

For me v6 make some usecases a little bit difficult.
For example I want to mark a Tab in my form as red when some specific fields are invalid. Etc.

How do you experience the new architecture?

Beside that I really like redux-form thank you for your contributions!

**Comment by mihirsoni** [comment]
@aight8 I believe you can make use of  ``_error`` which is form level error that you can define. 
for more you can visit :- http://redux-form.com/6.0.0-alpha.4/docs/api/Props.md/

**Comment by aight8** [comment]
Yeah that's the generic form error I know that already. But I want to mark a specific tab when some fields under this tab are invalid. More generally: How can I determinate if one of some fields are invalid

**Comment by mihirsoni** [comment]
@aight8 So , you are talking about something like multi page (Wizard form) , in that case yes there is a bug in v6. I have added the example and @erikras  has confirmed that as well [here](https://github.com/erikras/redux-form/pull/850#issuecomment-213690778)

**Comment by erikras** [comment]
Yes, the wizard form validation thing should be fixed tomorrow with a new alpha release.

**Comment by mihirsoni** [comment]
@aight8 This should be fixed now , you can validate at http://redux-form.com/6.0.0-alpha.6/examples/wizard/

Let us know if you were looking for the same approach.

**Comment by erikras** [comment]
> For example I want to mark a Tab in my form as red when some specific fields are invalid. Etc.

> How do you experience the new architecture?

This is related to #841. I have an idea of how to manage this. I'll see what I can do in a day or two.

**Comment by anyong** [comment]
I also have a question about errors, it worked well with alpha5 but stopped in alpha6. I use redux-form to manage all of my form state, but I fire all the actions myself using redux-saga to control the flow.

In alpha5, I simply called: `stopSubmit(formName, errors)` with my errors object:

```js
{
  field: 'error string'
}
```

and that worked fine for populating the error fields that look like this:

```js
{props.error && <span style={{color: 'red'}}>{props.error}</span>}
```

Now in alpha6 there is a `setSubmitFailed` action that takes a `...fields` parameter, and I'm not sure how to populate my error fields anymore. Can we keep the original behavior of `stopSubmit(formName, errors)` while still allowing whatever new behavior is provided by `setSubmitFailed`?

**Comment by erikras** [comment]
> If the new style is easier for internal use, do you think the publicly exported ones could be wrappers that use the original signatures? They are much more obvious to end users than a spread plus a few extra fields.

@anyong Is that not how they are exported? There's some trickery in `createAll.js` that should be adding the `form` and `field` params to the exported action creators.

**Comment by anyong** [comment]
Aha, I was missing the `setSubmitFailed` action from `index.js` so I just imported directly from `actions` and didn't get the `createAll` trickery.

I know you've been updating a lot so must have just missed that one ;)

**Comment by nktssh** [comment]
Hi all!
How can I import [change action creator](http://redux-form.com/6.0.0-alpha.7/docs/api/ActionCreators.md/)  from `redux-form v6.0.0-alpha.v7` ?
It seems that I have access to CHANGE action type, not to action creator...
